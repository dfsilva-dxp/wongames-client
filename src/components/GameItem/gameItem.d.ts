export type PaymentInfoProps = {
  number: string;
  flag: string;
  img: string;
  purchaseDate: string;
};

export type GameItemProps = {
  img: string;
  title: string;
  price: number;
  downloadLink?: string;
  paymentInfo?: PaymentInfoProps;
};
