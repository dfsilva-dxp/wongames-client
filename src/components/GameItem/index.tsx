import Image from "next/image";
import { Download } from "@styled-icons/boxicons-solid/Download";

import * as S from "./styles";

import { GameItemProps } from "./gameItem";

const GameItem = ({
  img,
  title,
  price,
  downloadLink,
  paymentInfo
}: GameItemProps) => {
  return (
    <S.Wrapper>
      <S.GameContent>
        <S.ImageBox>
          <Image src={img} title={title} alt={title} width={100} height={100} />
        </S.ImageBox>

        <S.Content>
          <S.Title>
            {title}
            {!!downloadLink && (
              <S.DownloadLink
                href={downloadLink}
                target="_blank"
                aria-label={`Get ${title} here`}
              >
                <Download size={22} />
              </S.DownloadLink>
            )}
          </S.Title>
          <S.Price>{price}</S.Price>
        </S.Content>
      </S.GameContent>

      {!!paymentInfo && (
        <S.PaymentContent>
          <p>{paymentInfo.purchaseDate}</p>
          <S.CardInfo>
            <span>{paymentInfo.number}</span>
            <Image
              src={paymentInfo.img}
              alt={paymentInfo.flag}
              width={100}
              height={100}
            />
          </S.CardInfo>
        </S.PaymentContent>
      )}
    </S.Wrapper>
  );
};

export default GameItem;
