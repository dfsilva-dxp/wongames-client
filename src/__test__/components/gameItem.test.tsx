import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import GameItem from "components/GameItem";

const props = {
  img: "http://localhost:1337/uploads/Image_9620999b16.png",
  title: "Red Dead Redemption 2",
  price: 215.0
};

describe("<GameItem />", () => {
  it("should render the item", () => {
    const { container } = renderWithTheme(<GameItem {...props} />);

    expect(
      screen.getByRole("heading", { name: props.title })
    ).toBeInTheDocument();

    expect(screen.getByRole("img", { name: props.title })).toBeInTheDocument();

    expect(screen.getByText("215")).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the item with download links", () => {
    const downloadLink = "http://link";

    renderWithTheme(<GameItem {...props} downloadLink={downloadLink} />);

    expect(
      screen.getByRole("link", { name: `Get ${props.title} here` })
    ).toHaveAttribute("href", downloadLink);
  });

  it("should render the payment info", () => {
    const paymentInfo = {
      flag: "mastercard",
      img: "/img/master-card.png",
      number: "**** **** **** 4326",
      purchaseDate: "Purchase made on 07/20/2020 at 20:32"
    };

    renderWithTheme(<GameItem {...props} paymentInfo={paymentInfo} />);

    expect(
      screen.getByRole("img", { name: paymentInfo.flag })
    ).toBeInTheDocument();

    expect(screen.getByText(paymentInfo.number)).toBeInTheDocument();

    expect(screen.getByText(paymentInfo.purchaseDate)).toBeInTheDocument();
  });
});
