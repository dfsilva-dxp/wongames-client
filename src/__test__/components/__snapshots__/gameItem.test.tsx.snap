// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameItem /> should render the item 1`] = `
.c0 {
  border: 1px solid #fff;
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
  overflow: hidden;
}

.c2 img {
  width: 100%;
  -webkit-block-size: cover;
  -ms-flex-block-size: cover;
  block-size: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c5 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="Red Dead Redemption 2"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=http%3A%2F%2Flocalhost%3A1337%2Fuploads%2FImage_9620999b16.png&w=256&q=75"
        srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A1337%2Fuploads%2FImage_9620999b16.png&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Flocalhost%3A1337%2Fuploads%2FImage_9620999b16.png&w=256&q=75 2x"
        style="color: transparent;"
        title="Red Dead Redemption 2"
        width="100"
      />
    </div>
    <div
      class="c3"
    >
      <h3
        class="c4"
      >
        Red Dead Redemption 2
      </h3>
      <div
        class="c5"
      >
        215
      </div>
    </div>
  </div>
</div>
`;
